web-common/src/features/dashboards/url-state/convert-partial-explore-state-to-url-params.ts

```diff
@@ -0,0 +1,385 @@
+import { mergeDimensionAndMeasureFilters } from "@rilldata/web-common/features/dashboards/filters/measure-filters/measure-filter-utils";
+import {
+  type PivotChipData,
+  PivotChipType,
+} from "@rilldata/web-common/features/dashboards/pivot/types";
+import { SortDirection } from "@rilldata/web-common/features/dashboards/proto-state/derived-types";
+import type { MetricsExplorerEntity } from "@rilldata/web-common/features/dashboards/stores/metrics-explorer-entity";
+import type { TimeControlState } from "@rilldata/web-common/features/dashboards/time-controls/time-control-store";
+import {
+  compressUrlParams,
+  shouldCompressParams,
+} from "@rilldata/web-common/features/dashboards/url-state/compression";
+import { convertExpressionToFilterParam } from "@rilldata/web-common/features/dashboards/url-state/filters/converters";
+import { FromLegacySortTypeMap } from "@rilldata/web-common/features/dashboards/url-state/legacyMappers";
+import {
+  ExploreUrlWebView,
+  FromActivePageMap,
+  ToURLParamSortTypeMap,
+  ToURLParamTDDChartMap,
+  ToURLParamTimeDimensionMap,
+  ToURLParamTimeGrainMapMap,
+  ToURLParamViewMap,
+} from "@rilldata/web-common/features/dashboards/url-state/mappers";
+import {
+  ExploreStateKeyToURLParamMap,
+  ExploreStateURLParams,
+} from "@rilldata/web-common/features/dashboards/url-state/url-params";
+import { arrayOrderedEquals } from "@rilldata/web-common/lib/arrayUtils";
+import {
+  TimeComparisonOption,
+  type TimeRange,
+  TimeRangePreset,
+} from "@rilldata/web-common/lib/time/types";
+import { copyParamsToTarget } from "@rilldata/web-common/lib/url-utils";
+import { DashboardState_ActivePage } from "@rilldata/web-common/proto/gen/rill/ui/v1/dashboard_pb";
+import { type V1ExploreSpec } from "@rilldata/web-common/runtime-client";
+
+export function convertPartialExploreStateToUrlParams(
+  exploreSpec: V1ExploreSpec,
+  partialExploreState: Partial<MetricsExplorerEntity>,
```

Comment:
In the future, I think this should be a `exploreState.toURLSearchParams()` function that operates on a complete `ExploreState` not a partial one.

Reply:
It is currently used in `updateExploreSessionStore` where it could be a partial. So lets do this in a follow up.